src/saltext/vmware/modules/esxi.py

```diff
@@ -732,3 +732,131 @@ def set_advanced_config(
         host_name=host_name,
         service_instance=service_instance,
     )
+
+
+def connect(name, service_instance=None):
+    """
+    Connect an ESXi instance to a vCenter instance.
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance from which to obtain managed object references. (Optional)
+    """
+    log.debug(f"Connect ESXi instance {name}.")
+    if service_instance is None:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+
+    state = utils_esxi.reconnect_host(name, service_instance)
+    return {"state": state}
+
+
+def disconnect(name, service_instance=None):
+    """
+    Disconnect an ESXi instance.
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance from which to obtain managed object references. (Optional)
+    """
+    log.debug(f"Disconnect ESXi instance {name}.")
+    if service_instance is None:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+
+    state = utils_esxi.disconnect_host(name, service_instance)
+    return {"state": state}
+
+
+def remove(name, service_instance=None):
+    """
+    Remove an ESXi instance from a vCenter instance.
+
+    name
+        Name of host.
+
+    service_instance
+        The Service Instance from which to obtain managed object references. (Optional)
+    """
+    log.debug(f"Remove ESXi instance {name}.")
+    if service_instance is None:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+
+    state = utils_esxi.remove_host(name, service_instance)
+    return {"state": state}
+
+
+def move(name, cluster_name, service_instance=None):
+    """
+    Move an ESXi instance to a different cluster.
+
+    name
+        Name of host.
+
+    cluster_name
+        Name of cluster to move host to.
+
+    service_instance
+        The Service Instance from which to obtain managed object references. (Optional)
+    """
+    log.debug(f"Move ESXi instance {name}.")
+    if service_instance is None:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+
+    state = utils_esxi.move_host(name, cluster_name, service_instance)
+    return {"state": state}
+
+
+def add(
+    ip,
+    root_user,
+    password,
+    cluster_name,
+    datacenter_name,
+    server_auth=True,
```

Comment:
I'm not sure I like `server_auth` as a name and its description in the body.

The fingerprint auth always happens, what we're verifying is that the hostname matches its certificate, which is signed by a certificate authority (I think?).

Reply:
Something had bit nagging at the back of my mind and I realize what it was - do we need to do this kind of validation ourselves or is it something that we can delegate to, say, pyvmomi or requests or something? :thinking: